runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -78,7 +78,7 @@ J9::CodeGenerator::CodeGenerator() :
    _liveMonitors(NULL),
    _nodesSpineCheckedList(getTypedAllocator<TR::Node*>(TR::comp()->allocator())),
    _jniCallSites(getTypedAllocator<TR_Pair<TR_ResolvedMethod,TR::Instruction> *>(TR::comp()->allocator())),
-   _monitorMapping(self()->comp()->trMemory(), 256),
+   _monitorMapping(std::less<ncount_t>(), MonitorMapAllocator(self()->comp()->trMemory()->heapMemoryRegion())),
```

Comment:
Sorry, I missed this earlier: you should avoid calling `self()` in constructors because it relies on undefined behaviour. You can use `TR::comp()` instead as is done to initialize other fields.
```suggestion
   _monitorMapping(std::less<ncount_t>(), MonitorMapAllocator(TR::comp()->trMemory()->heapMemoryRegion())),
```

Reply:
fixed as suggested. thanks!